dp3t-sdk/sdk/src/androidTest/java/org/dpppt/android/sdk/internal/backend/SignatureVerificationInterceptorTest.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) 2020 Ubique Innovation AG <https://www.ubique.ch>
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package org.dpppt.android.sdk.internal.backend;
+
+import android.content.Context;
+import androidx.test.platform.app.InstrumentationRegistry;
+
+import java.io.IOException;
+import java.security.PublicKey;
+
+import org.dpppt.android.sdk.backend.SignatureException;
+import org.dpppt.android.sdk.internal.logger.LogLevel;
+import org.dpppt.android.sdk.internal.logger.Logger;
+import org.dpppt.android.sdk.models.DayDate;
+import org.dpppt.android.sdk.util.SignatureUtil;
+import org.junit.Before;
+import org.junit.Test;
+
+import okhttp3.mockwebserver.Dispatcher;
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.MockWebServer;
+import okhttp3.mockwebserver.RecordedRequest;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+public class SignatureVerificationInterceptorTest {
+
+	Context context;
+	MockWebServer server;
+	BackendBucketRepository bucketRepository;
+
+	@Before
+	public void setup() {
+		context = InstrumentationRegistry.getInstrumentation().getContext();
+
+		Logger.init(context, LogLevel.DEBUG);
+
+		ProxyConfig.DISABLE_SYSTEM_PROXY = true;
+
+		server = new MockWebServer();
+		PublicKey signaturePublicKey = SignatureUtil.getPublicKeyFromBase64OrThrow(
+				"LS0tLS1CRUdJTiBQVUJMSUMgS0VZLS0tLS0KTUZrd0V3WUhLb1pJemowQ0FRWUlLb1pJemowREFRY0RRZ0FFaXFSZ2FvYzdMb0p" +
+						"jdUx3d3F1OGszNmhVc2dheQp1a0lTR2p2cEtab05vNGZRNWJsekFUV3VBK0E4eklDRnFDOFNXQmlvZkFCRmxqandNeDR2ejlobGVnPT0KL" +
+						"S0tLS1FTkQgUFVCTElDIEtFWS0tLS0t");
+		bucketRepository = new BackendBucketRepository(context, server.url("/bucket/").toString(), signaturePublicKey);
+	}
+
+	@Test
+	public void testValidSignature() throws IOException, StatusCodeException {
+		String responseString = "{\"forceUpdate\":false,\"forceTraceShutdown\":false,\"infoBox\":null," +
+				"\"sdkConfig\":{\"numberOfWindowsForExposure\":3,\"eventThreshold\":0.8,\"badAttenuationThreshold\":73.0," +
+				"\"contactAttenuationThreshold\":73.0},\"iOSGaenSdkConfig\":{\"lowerThreshold\":55,\"higherThreshold\":63," +
+				"\"factorLow\":1.0,\"factorHigh\":0.5,\"triggerThreshold\":15},\"androidGaenSdkConfig\":{\"lowerThreshold\":55," +
+				"\"higherThreshold\":63,\"factorLow\":1.0,\"factorHigh\":0.5,\"triggerThreshold\":15}}";
+		server.setDispatcher(new Dispatcher() {
+			@Override
+			public MockResponse dispatch(RecordedRequest request) {
+				return new MockResponse()
+						.setResponseCode(200)
+						.setBody(responseString)
+						.addHeader(SignatureUtil.HTTP_HEADER_JWS,
+								"eyJhbGciOiJFUzI1NiJ9" +
```

Comment:
This token expires on October 5th 2020. After that the test will fail. 
We might need some JWS generator?